lib/bike_brigade_web/components/conversation_component.ex

```diff
@@ -0,0 +1,135 @@
+defmodule BikeBrigadeWeb.Components.ConversationComponent do
+  use BikeBrigadeWeb, :live_component
+
+  alias BikeBrigade.{Messaging, MediaStorage, SmsService, Messaging.SmsMessage}
+
+  defdelegate campaign_name(campaign), to: CampaignHelpers, as: :name
+
+  @impl Phoenix.LiveComponent
+  def mount(socket) do
+    socket =
+      socket
+      |> allow_upload(:media, accept: ~w(.gif .png .jpg .jpeg), max_entries: 10)
+      |> assign_confirm_send_warning()
+
+    {:ok, socket, temporary_assigns: [scrollback: [], conversation: []]}
+  end
+
+  def assign_confirm_send_warning(socket) do
+    assign(socket, :confirm_send, SmsService.sending_confirmation_message())
+  end
+
+  @impl Phoenix.LiveComponent
+  def update(%{rider: rider} = assigns, socket) do
+    conversation = Messaging.latest_messages(rider)
+
+    earliest_timestamp =
+      case conversation do
+        [first | _] -> first.sent_at
+        [] -> nil
+      end
+
+    {:ok,
+     socket
+     |> assign(assigns)
+     |> assign_new_sms_message()
+     |> assign(conversation: conversation)
+     |> assign(earliest_timestamp: earliest_timestamp)
+     |> assign(phx_update: "append")}
```

Comment:
Let's pair on getting this to use streams! (outside this PR)

Reply:
Yes, 💯 !